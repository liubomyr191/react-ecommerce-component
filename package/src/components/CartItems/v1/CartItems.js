import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { withComponents } from "@reactioncommerce/components-context";
import { CustomPropTypes } from "../../../utils";

const Items = styled.div``;

class CartItems extends Component {
  static propTypes = {
    /**
     * If you've set up a components context using @reactioncommerce/components-context
     * (recommended), then this prop will come from there automatically. If you have not
     * set up a components context or you want to override one of the components in a
     * single spot, you can pass in the components prop directly.
     */
    components: PropTypes.shape({
      /**
       * Pass either the Reaction `CartItem` component or your own component
       * that takes `items`, `isMiniCart`, `onChangeCartItemQuantity`, and
       * `onRemoveItemFromCart` props and uses them to render a single cart item.
       */
      CartItem: CustomPropTypes.component.isRequired
    }).isRequired,
    /**
     * Is in a MiniCart component
     */
    isMiniCart: PropTypes.bool,
    /**
     * Hide remove button and quantity input
     */
    isReadOnly: PropTypes.bool,
    /**
     * CartItem data. Only the `_id` prop is required by this component. Each item is passed to
     * CartItem, which may require additional props.
     */
    items: PropTypes.arrayOf(PropTypes.shape({
      /**
       * The cart item ID
       */
      _id: PropTypes.string.isRequired
    })).isRequired,
    /**
     * On cart item quantity change handler
     */
    onChangeCartItemQuantity: PropTypes.func,
    /**
     * On remove item from cart handler
     */
    onRemoveItemFromCart: PropTypes.func
  };

  static defaultProps = {
    isMiniCart: false,
    isReadOnly: false,
    onChangeCartItemQuantity() {},
    onRemoveItemFromCart() {}
  };

  render() {
    const { items, components: { CartItem, ...components }, ...props } = this.props;
    return (
      <Items>
        {items.map((item) => <CartItem key={item._id} item={item} components={components} {...props} />)}
      </Items>
    );
  }
}

export default withComponents(CartItems);
