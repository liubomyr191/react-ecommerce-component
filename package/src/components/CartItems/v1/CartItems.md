### CartItems

### Overview
This component renders a list of items that are in a customers cart.

#### Basic Usage
```jsx
const items = [{
  _id: "123",
  attributes: [{ label: "vendor", value: "Patagonia" }, { label: "Color", value: "Red" }, { label: "Size", value: "Medium" }],
  currentQuantity: 3,
  imageUrl: "//placehold.it",
  isLowInventoryQuantity: true,
  price: {
    displayPrice: "$20.00",
    compareAtPrice: "$45.00"
  },
  productSlug: "/product-slug",
  title: "A Great Product",
  quantity: 2
},
{
  _id: "456",
  attributes: [{ label: "vendor", value: "Nike" }, { label: "Color", value: "Black" }, { label: "Size", value: "10" }],
  currentQuantity: 500,
  imageUrl: "//placehold.it",
  isLowInventoryQuantity: false,
  price: {
    displayPrice: "$78.00"
  },
  productSlug: "/product-slug",
  title: "Another Great Product",
  quantity: 1
}];

const components = {
  CartItemComponent: CartItem,
  CartItemDetailComponent: CartItemDetail,
  CartItemStockWarningComponent: StockWarning,
  CartItemPriceComponent: Price,
  CartItemQuantityInputComponent: QuantityInput
};

<CartItems items={items} components={components} onChangeCartItemQuantity={(value) => console.log("cart items new quantity", value)} onRemoveItemFromCart={(_id) => console.log("cart items remove this item", _id)}/>
```


#### In Mini Cart
```jsx
const items = [{
  _id: "123",
  attributes: [{ label: "vendor", value: "Patagonia" }, { label: "Color", value: "Red" }, { label: "Size", value: "Medium" }],
  currentQuantity: 3,
  imageUrl: "//placehold.it",
  isLowInventoryQuantity: true,
  price: {
    displayPrice: "$20.00",
    compareAtPrice: "$45.00"
  },
  productSlug: "/product-slug",
  title: "A Great Product",
  quantity: 2
},
{
  _id: "456",
  attributes: [{ label: "vendor", value: "Nike" }, { label: "Color", value: "Black" }, { label: "Size", value: "10" }],
  currentQuantity: 500,
  imageUrl: "//placehold.it",
  isLowInventoryQuantity: false,
  price: {
    displayPrice: "$78.00"
  },
  productSlug: "/product-slug",
  title: "Another Great Product",
  quantity: 1
}];

const components = {
  CartItemComponent: CartItem,
  CartItemDetailComponent: CartItemDetail,
  CartItemStockWarningComponent: StockWarning,
  CartItemPriceComponent: Price,
  CartItemQuantityInputComponent: QuantityInput
};

<CartItems items={items} components={components} onChangeCartItemQuantity={(value) => console.log("cart items new quantity", value)} onRemoveItemFromCart={(_id) => console.log("cart items remove this item", _id)} isMiniCart />
```
