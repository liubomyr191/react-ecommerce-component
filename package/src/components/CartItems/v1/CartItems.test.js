import React from "react";
import renderer from "react-test-renderer";
import CartItems from "./CartItems";

test("basic snapshot with empty props", () => {
  const component = renderer.create(<CartItems items={[]} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot with props", () => {
  const mockItems = [
    {
      _id: "123",
      attributes: [
        { label: "vendor", value: "Patagonia" },
        { label: "Color", value: "Red" },
        { label: "Size", value: "Medium" }
      ],
      currentQuantity: 3,
      imageUrl: "http://placehold.it",
      isLowInventoryQuantity: true,
      price: {
        displayPrice: "$20.00",
        compareAtPrice: "$45.00"
      },
      productSlug: "/product-slug",
      title: "A Great Product",
      quantity: 2
    },
    {
      _id: "456",
      attributes: [
        { label: "vendor", value: "Nike" },
        { label: "Color", value: "Black" },
        { label: "Size", value: "10" }
      ],
      currentQuantity: 500,
      imageUrl: "http://placehold.it",
      isLowInventoryQuantity: false,
      price: {
        displayPrice: "$78.00"
      },
      productSlug: "/product-slug",
      title: "Another Great Product",
      quantity: 1
    }
  ];
  const mockComponents = {
    CartItemComponent: () => "Cart Item",
    CartItemDetailComponent: () => "Cart Item Detail",
    CartItemStockWarningComponent: () => "Cart Item Stock Warning",
    CartItemPriceComponent: () => " Cart Item Price",
    CartItemQuantityInputComponent: () => "Cart Item Quantity Input"
  };

  const component = renderer.create(<CartItems items={mockItems} components={mockComponents} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
