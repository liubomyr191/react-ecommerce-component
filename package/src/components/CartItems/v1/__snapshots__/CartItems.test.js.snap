// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot with empty props 1`] = `
<div
  className=""
/>
`;

exports[`basic snapshot with props 1`] = `
<div
  className=""
>
  CartItem(undefined)
  CartItem(undefined)
</div>
`;
